tests/test_git.py

```diff
@@ -1041,3 +1041,50 @@ def test_status(
     assert staged["modify"] == ["foo"]
     assert unstaged == ["bar"]
     assert untracked == expected_untracked
+
+
+@pytest.mark.skip_git_backend("pygit2")
+@pytest.mark.parametrize("untracked_files", [False, True])
+def test_is_dirty(
+    tmp_dir: TmpDir,
+    scm: Git,
+    git: Git,
+    tmp_dir_factory: TempDirFactory,
+    mocker,
+    untracked_files: str,
+):
+    tmp_dir.gen(
+        {
+            "foo": "foo",
+        }
+    )
+    scm.add_commit(["foo"], message="init")
+
+    assert not git.is_dirty()
+
+    with (tmp_dir / "foo").open("a") as fobj:
+        fobj.write("modified")
+
+    assert git.is_dirty(untracked_files)
+    scm.add_commit(["foo"], message="commit")
+
+    with (tmp_dir / "untracked").open("w") as fobj:
+        fobj.write("untracked")
+
+    backend = list(git.backends)[0]
+    assert backend in ("dulwich", "gitpython")
+    if backend == "dulwich":
+        spy = mocker.spy(git.dulwich, "status")
+    elif backend == "gitpython":
+        spy = mocker.spy(git.gitpython.repo, "is_dirty")
```

Comment:
I don't think we need to be `spy`ing individual backend calls at all here. `test_git` tests should just be asserting whether or the relevant call (`is_dirty()` here) returns the expected results across each backend.

Reply:
Yeah, this is pretty ugly. I'd probably rather go to having one test per backend. There will be some setup-related duplicated code, but the tests themselves will be less confusing.

Reply:
@dtrifiro Having two separate tests in `test_git.py` makes it look like we expect different results for `is_dirty()` depending on which backend we are using. IMO `test_git` should have one test that verifies that `is_dirty()` returns the correct values regardless of backend (and it should not test any backend-specific behavior).

I'm fine with adding a separate test for verifying the dulwich kwargs in `test_dulwich.py`, but I don't think that the gitpython specific test behavior adds anything beneficial (since gitpython's native `is_dirty` doesn't have the distinct options for different levels of untracked files behavior)

Reply:
I ended up scrapping the `spy`ing to keep things simpler, as the value of that check was questionable